src/mlpack/core/data/load_image_impl.hpp

```diff
@@ -0,0 +1,210 @@
+             width,
+             height,
+             channels,
+             STBI_grey);
+  }
+  else
+  {
+    image = stbi_load(fileName.c_str(),
+             width,
+             height,
+             channels,
+             STBI_rgb);
+  }
+
+  if (*width <= 0 || *height <=0)
+  {
+    std::ostringstream oss;
+    oss << "Image '" << fileName << "' not found." << std::endl;
+    free(image);
+    // throw std::runtime_error(oss.str());
+    return false;
+  }
+
+  int size = (*width)*(*height)*(*channels);
+
+  // Copy memory location into armadillo Mat.
+  outputMatrix = arma::Mat<unsigned char>(image, 1, size, false, true);
+  return true;
+}
+
+bool LoadImage::Load(const std::string& fileName,
+                    bool flipVerticalOnLoad,
+                    arma::Mat<unsigned char>&& outputMatrix)
+{
+  int width, height;
+  bool status = Load(fileName,
+      flipVerticalOnLoad,
+      std::move(outputMatrix),
+      &width, &height, &channels);
+  if (!status)
+    return status;
+
+  Log::Info << width <<" "<< height <<" "<< channels << "\n" << std::flush;
+
+  // Throw error if the image is incompatible with the matrix.
+  if (matrixWidth > 0 && matrixHeight > 0 &&
+     (width != matrixWidth || height != matrixHeight))
+  {
+    std::ostringstream oss;
+    oss << "Image '" << fileName;
+    oss << "' does not match matrix height or width." << std::endl;
+    oss << "Image width: " << width;
+    oss << ", Image height: " << height << std::endl;
+    oss << "Matrix Width: " << matrixWidth;
+    oss << ", Matrix Height: " << matrixHeight << std::endl;
+    throw std::runtime_error(oss.str());
+  }
+  return status;
+}
+
+bool LoadImage::Load(std::vector<std::string>& files,
+                     bool flipVerticalOnLoad,
+                     arma::Mat<unsigned char>&& outputMatrix)
+{
+  if (files.size() < 1)
+  {
+    std::ostringstream oss;
+    oss << "File vector is empty." << std::endl;
+    throw std::runtime_error(oss.str());
+  }
+
+  arma::Mat<unsigned char> img;
+  int width, height;
+  bool status = Load(files[0],
+                     flipVerticalOnLoad,
+                     std::move(img),
+                     &width, &height, &channels);
+  Log::Info << "Loaded " << files[0] << std::endl << std::flush;
+
+  // Decide matrix dimension using the image height and width.
+  matrixWidth = std::max(matrixWidth, width);
+  matrixHeight = std::max(matrixHeight, height);
+
+  outputMatrix.set_size(files.size(), matrixWidth*matrixHeight*channels);
+  outputMatrix.row(0) = img;
+
+  for (size_t i = 1; i < files.size() ; i++)
+  {
+    status &= Load(files[i],
+                   flipVerticalOnLoad,
+                   std::move(outputMatrix.row(i)));
+    Log::Info << "Loaded " << files[i] << std::endl << std::flush;
+  }
+  return status;
+}
+
+bool LoadImage::LoadDIR(const std::string& dirPath,
+                        bool flipVerticalOnLoad,
+                        arma::Mat<unsigned char>&& outputMatrix)
```

Comment:
Do you think we can use a reference here, it might be confusing for the user to use a different API when loading images. Let me know what you think.